src/tss2-sys/api/Tss2_Sys_GetCapability.c

```diff
@@ -75,7 +78,31 @@ TSS2_RC Tss2_Sys_GetCapability_Complete(
     if (rval)
         return rval;
 
-    return Tss2_MU_TPMS_CAPABILITY_DATA_Unmarshal(ctx->cmdBuffer,
+    /* Peak at the capabilityData->capability field to decide what to do
+       if this is a standard cap, just unmarshal it.
+       if this is a vendor cap, we fill a tpm2b inside the struct buffer.*/
+    rval = Tss2_MU_UINT32_Unmarshal(&ctx->cmdBuffer[ctx->nextData],
+                                  ctx->maxCmdSize - ctx->nextData,
+                                  NULL,
+                                  &cap);
+    if (rval)
+        return rval;
+
+    /* Size the capabilityData is the end of response we can copy the remainder
+       of the response in cas of cap_vendor */
+    if (cap == TPM2_CAP_VENDOR_PROPERTY) {
+        if (ctx->maxCmdSize - ctx->nextData > sizeof(capabilityData->data.vendor.buffer))
+            return TSS2_MU_RC_BAD_SIZE;
+
+        capabilityData->capability = cap;
+        capabilityData->data.vendor.size = ctx->maxCmdSize - ctx->nextData;
+        memcpy(&capabilityData->data.vendor.buffer[0],
+               &ctx->cmdBuffer[ctx->nextData],
+               capabilityData->data.vendor.size);
+        ctx->nextData = ctx->maxCmdSize;
+        return TSS2_RC_SUCCESS;
+    } else
```

Comment:
No need for the else with return. Even though we lose the tail call optimization, I would be in favor of the flow being:

```C
if (cap != TPM2_CAP_VENDOR_PROPERTY)
        return Tss2_MU_TPMS_CAPABILITY_DATA_Unmarshal(ctx->cmdBuffer,
                                                  ctx->maxCmdSize,
                                                  &ctx->nextData,
                                                  capabilityData);

/* now do complicated vendor stuff */
...
```